src/arch/src/aarch64/regs.rs

```diff
@@ -139,9 +144,51 @@ macro_rules! arm64_sys_reg {
     };
 }
 
+// This macro extracts the MANUFACTURER_ID from the provided MIDR_EL1 register value,
+// found between bits 24 and 31.
+// https://developer.arm.com/documentation/100095/0001/system-control/
+// aarch64-register-summary/aarch64-identification-registers?lang=en
+macro_rules! manufacturer_id {
```

Comment:
I would avoid using a macro since it can be substitute with one line.

Reply:
So basically directly computing >> 24